File: Queue/QueueMethods/que.c
#include <stdio.h>
#include <stdbool.h>
#define MAX 6

int intArray[MAX];
int front = 0;
int rear = -1;
int itemCount = 0;

bool isFull() { return itemCount == MAX; }
bool isEmpty() { return itemCount == 0; }

void enqueue(int data) {
    if (!isFull()) {
        if (rear == MAX - 1) rear = -1;
        intArray[++rear] = data;
        itemCount++;
    } else {
        printf("Queue is full!\n");
    }
}

int dequeue() {
    if (!isEmpty()) {
        int data = intArray[front++];
        if (front == MAX) front = 0;
        itemCount--;
        return data;
    }
    printf("Queue is empty!\n");
    return -1;
}

int peek() {
    if (!isEmpty()) return intArray[front];
    printf("Queue is empty!\n");
    return -1;
}

void display() {
    if (isEmpty()) {
        printf("Queue is empty!\n");
        return;
    }
    printf("Queue: ");
    int count = itemCount, f = front;
    while (count--) {
        printf("%d ", intArray[f]);
        f = (f + 1) % MAX;
    }
    printf("\n");
}

int main() {
    enqueue(3);
    enqueue(5);
    enqueue(9);
    enqueue(1);
    enqueue(12);
    enqueue(15);

    display();
    printf("Front (peek): %d\n", peek());

    printf("Removed: %d\n", dequeue());
    display();
    printf("Front after dequeue (peek): %d\n", peek());
}

File: Queue/QueueMethods/que.py
MAX = 6

class Queue:
    def __init__(self):
        self.arr = [0] * MAX
        self.front = 0
        self.rear = -1
        self.itemCount = 0

    def isFull(self):
        return self.itemCount == MAX

    def isEmpty(self):
        return self.itemCount == 0

    def enqueue(self, data):
        if not self.isFull():
            if self.rear == MAX - 1:
                self.rear = -1
            self.rear += 1
            self.arr[self.rear] = data
            self.itemCount += 1
        else:
            print("Queue is full!")

    def dequeue(self):
        if not self.isEmpty():
            data = self.arr[self.front]
            self.front += 1
            if self.front == MAX:
                self.front = 0
            self.itemCount -= 1
            return data
        print("Queue is empty!")
        return -1

    def peek(self):
        if not self.isEmpty():
            return self.arr[self.front]
        print("Queue is empty!")
        return -1

    def display(self):
        if self.isEmpty():
            print("Queue is empty!")
            return
        print("Queue:", end=" ")
        count = self.itemCount
        f = self.front
        while count > 0:
            print(self.arr[f], end=" ")
            f = (f + 1) % MAX
            count -= 1
        print()

# Example
q = Queue()
q.enqueue(3)
q.enqueue(5)
q.enqueue(9)
q.enqueue(1)
q.enqueue(12)
q.enqueue(15)

q.display()
print("Front (peek):", q.peek())

print("Removed:", q.dequeue())
q.display()
print("Front after dequeue (peek):", q.peek())
